helix-core/src/main/java/org/apache/helix/model/CloudConfig.java

```diff
@@ -90,7 +86,7 @@ public CloudConfig(String cluster, boolean enabled, CloudProvider cloudProvider,
    * Enable/Disable the CLOUD_ENABLED field.
    * @param enabled
    */
-  public void setCloudEnabled(boolean enabled) {
+  private void setCloudEnabled(boolean enabled) {
```

Comment:
These methods are/can not used by any place. Instead of removing all, we can keep some of the them as public ones. We need to leave the room to let user change some of the fields, such as cloudenable, source fields.

For validation purpose, we can make all constructors private and creating through Builder.

Reply:
Fixed. I made 3 of them that users might use as public. I added a test for those methods as well.